cityofaustin/atd-moped
# Work around Mapbox's clipped geojson features (#6)

## Associated issues
https://github.com/cityofaustin/atd-data-tech/issues/10402

This PR updates the component draft mode to capture geometry from the same AGOL data that is populated in the map to show what lines and points are available to add to a component. Previously, we grabbed that geometry from a click event in the map library which led to the possibility of clipping geometries. Now, we click a feature, get the feature's unique ID, and then find the record within the features returned from AGOL to use its geometry and properties to eventually persist the component.

## Testing
**URL to test:** 

https://md-10402-clipped-geojson--atd-moped-main.netlify.app/moped/

**Steps to test:**

**This PR update doesn't change the way the UI works so this is mostly verifying that points and lines are still selectable.**

1. Go to a new or existing project's **Map** tab
2. Create a new component with the type of **Access Control - Driveway Closure** and click **Continue**
3. Select some line segments and click **Save** to add the component
4. Repeat with a second component with type of **Bike box** but select some points this time and then **Save**
5. You should be able to select the components that you created from the sidebar menu or click on them on the map and then click the button in the popup to highlight a component
6. Check to make sure that you cannot click the greyed out components when highlighting a component

---
#### Ship list
